src/tasks/receipts.rs

```diff
@@ -0,0 +1,74 @@
+use crate::config::Provider;
+use alloy::{primitives::TxHash, providers::Provider as _};
+use metrics::{counter, histogram};
+use std::time::Instant;
+use tokio::{sync::mpsc, task::JoinHandle};
+use tracing::{debug, error};
+
+/// Submits sidecars in ethereum txns to mainnet ethereum
+#[derive(Debug, Clone)]
+pub struct ReceiptTask {
+    /// Ethereum Provider
+    pub host_provider: Provider,
+}
+
+impl ReceiptTask {
+    pub async fn log_tx(&self, pending_tx_hash: TxHash) {
+        // start timer when tx hash is received
+        let start: Instant = Instant::now();
+
+        // wait for the tx to mine, get its receipt
+        let receipt = self.host_provider.clone().get_transaction_receipt(pending_tx_hash).await;
+
+        match receipt {
+            Ok(receipt) => {
```

Comment:
mostly a pedantic naming change, to make clear that this is a none type that can be none in the instance the tx simply isn't found.
```suggestion
            Ok(maybe_receipt) => {
```

Reply:
https://github.com/init4tech/builder/pull/48/commits/7ebd8d219556a9fa66d2c3c697b445c254d6029c